### ISSUE TITLE
[Feat][Checkbox]: Checkbox should support label, labelledby and describedby

### ISSUE THREAD

**Issue by majornista** [adobe/spectrum-web-components#3269]
### Code of conduct

- [X] I agree to follow this project's code of conduct.

### Impacted component(s)

Checkbox

### Description of the requested feature

It should be possible to label a Checkbox, without a visible label, using `label`, `labelledby` and `describedby`, so that for example checkboxes in a Table used for row selection can be labeled using `label="Select"` and `labelledby` referencing the checkbox itself and IDREFs for the row header cells.

**Comment by Westbrook** [comment]
Spectrum doesn't _technically_ support this usage: https://spectrum.adobe.com/page/checkbox/, which is why we don't support it. It's an interesting feature, and one that will be discussed more broadly as part of https://github.com/Igalia/webengineshackfest/issues/10 which looks to be a great conversation, amongst a bevy of great work at that event.

**Comment by majornista** [comment]
> Spectrum doesn't _technically_ support this usage: https://spectrum.adobe.com/page/checkbox/, which is why we don't support it. It's an interesting feature, and one that will be discussed more broadly as part of [Igalia/webengineshackfest#10](https://github.com/Igalia/webengineshackfest/issues/10) which looks to be a great conversation, amongst a bevy of great work at that event.

It looks like we should at least support `label`: https://spectrum.adobe.com/page/checkbox/#Table-of-options

Also, do we not support a `name` attribute for form submission?

**Comment by Westbrook** [comment]
We currently support `label` as text:
```html
<sp-checkbox>${label}</sp-checkbox>
```

**Comment by majornista** [comment]
But, we need to be able to label a standalone checkbox, as in this example: https://spectrum.adobe.com/page/checkbox/#When-to-use-a-standalone-checkbox, in order to disambiguate between different instances of a standalone checkbox.

![Standalone Checkbox example in Spectrum](https://github.com/adobe/spectrum-web-components/assets/154077/9513d861-82ab-4eff-8e34-52f04a0cdc3e)

**Comment by Westbrook** [comment]
Interesting. I'd want to know more about that pattern, and the accessibility model for it, before committing to a specific set of work in this area.

**Comment by majornista** [comment]
It would be equivalent to `<input type="checkbox" aria-label="Fill" checked />` or `<input type="checkbox" aria-label="Border" checked />`. If I were to use `<sp-checkbox checked></sp-checkbox>` for each of these checkboxes, there would be no way for someone using a screen reader to tell them apart.

**Comment by Westbrook** [comment]
Oh, I'm more wondering about the extended UI. I see this as [label content](https://studio.webcomponents.dev/edit/collection/fO75441E1Q5ZlI0e9pgq/jeIGAXHMUrTp6hGMquoD/src/index.ts?branch=swatch-box%40wfVpXdsTYhf9mY2slxW4t2Y2SjC3&p=stories), for the first two:
```html
<sp-checkbox><sp-swatch></sp-swatch>Fill</sp-checkbox>
```
But the last is unclear:
```html
<sp-checkbox>Background blue<sp-icon-chevron-down></sp-icon-chevron-down></sp-checkbox>
```
This delivers the UI, but what the accessibility tree _meant_ to be for such a UI?

**Comment by majornista** [comment]
> Oh, I'm more wondering about the extended UI. I see this as [label content](https://studio.webcomponents.dev/edit/collection/fO75441E1Q5ZlI0e9pgq/jeIGAXHMUrTp6hGMquoD/src/index.ts?branch=swatch-box%40wfVpXdsTYhf9mY2slxW4t2Y2SjC3&p=stories), for the first two:
> 
> ```html
> 
> <sp-checkbox><sp-swatch></sp-swatch>Fill</sp-checkbox>
> 
> ```
> 
> But the last is unclear:
> 
> ```html
> 
> <sp-checkbox>Background blue<sp-icon-chevron-down></sp-icon-chevron-down></sp-checkbox>
> 
> ```
> 
> This delivers the UI, but what the accessibility tree _meant_ to be for such a UI?

I assume what you have as a "swatch" here is a popup button for an interactive color picker, in which case, we shouldn't be nesting interactive content within the label for a Checkbox. Which is why we need a way to label the Checkbox without a visible label for this use case. 

We have ways to hide the label in the Slider control, and we can use `label` on Textfield to add an aria-label to the input.

**Comment by Rajdeepc** [comment]
Should wait for https://github.com/WICG/aom/blob/gh-pages/reference-target-explainer.md to land.
